include/device.h

```diff
@@ -286,6 +286,18 @@ struct device *device_get_binding(const char *name);
  */
 #define DEVICE_PM_SUSPEND_STATE         3
 
+/** @def DEVICE_PM_FORCE_SUSPEND_STATE
+ *
+ * @brief device is in force SUSPEND power state
+ *
+ * @details Driver puts the device in suspended state after
+ * terminating or completing the ongoing transactions and will
```

Comment:
IMHO we should never allow for terminating ongoing transaction as we could leave HW in undefined state.
However now I see what you would like to achieve. What about the following approach:

DEVICE_PM_SUSPEND_STATE:
 - Wait for completion of all ongoing and queued work, do not accept new requests.

DEVICE_PM_FORCE_SUSPEND_STATE:
 - Wait for completion of all ongoing work, cancel queued jobs (and notify upper layers about this fact!), do not accept new requests.